File: Core/Entities/Receivable.cs
namespace Core.Entities
{
    public class Receivable : BaseEntity
    {
        public decimal AmountDue { get; set; }
        public decimal AmountPaid { get; set; }
        public DateTime DateReceived { get; set; }
        public string InvoiceNumber { get; set; }
        public string Method { get; set; }
        public string Notes { get; set; }
        public int EntityId { get; set; }
        public Entity Entity { get; set; }
        public int GigId { get; set; }
        public Gig Gig { get; set; }
    }
}

File: client/src/app/shared/models/receivable.ts
import { IGig } from "./gig";

export interface IReceivable {
    id: number;
    amountDue: number;
    amountPaid: number;
    dateReceived: Date;
    invoiceNumber: string;
    method: string;
    notes: string;
    entityId: number;
    entity: string;
    gig: IGig;
}